src/app/patient/repositories/record.go

```diff
@@ -0,0 +1,70 @@
+package patient_repositories
+
+import (
+	"clinic-api/src/app/patient"
+	"clinic-api/src/types"
+	"time"
+
+	"github.com/google/uuid"
+	"gorm.io/gorm"
+)
+
+type Patient struct {
+	gorm.Model
+	ID        uuid.UUID `gorm:"primaryKey;size:191"`
+	Name      string
+	NIK       string
+	Phone     string
+	Address   string
+	DOB       time.Time
+	Gender    types.GenderEnum `gorm:"type:enum('MALE', 'FEMALE')"`
+	BloodType string
+}
+
+func (record *Patient) MapToDomain() patient.Domain {
+	return patient.Domain{
+		ID:        record.ID,
+		Name:      record.Name,
+		NIK:       record.NIK,
+		Phone:     record.Phone,
+		Address:   record.Address,
+		DOB:       record.DOB,
+		Gender:    record.Gender,
+		BloodType: record.BloodType,
+	}
+}
+
+func MapToNewRecord(domain patient.Domain) Patient {
+	return Patient{
+		ID:        uuid.Must(uuid.NewRandom()),
+		Name:      domain.Name,
+		NIK:       domain.NIK,
+		Phone:     domain.Phone,
+		Address:   domain.Address,
+		DOB:       domain.DOB,
+		Gender:    domain.Gender,
+		BloodType: domain.BloodType,
+	}
+}
+
+func MapToExistingRecord(domain patient.Domain) Patient {
+	return Patient{
+		ID:        uuid.Nil,

```

Comment:
Prefer not to specify this, cause it will be nil by default or you can use `.Omit()` in repositories to ignore the id when update

Reply:
Implemented at https://github.com/ALTERA-CAPSTON-41/backend/pull/38/commits/8e2b69069a1b40f3059ff9cc80045a2727151867.